lib/bundle_update_interactive/git_committer.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+require "shellwords"
+
+module BundleUpdateInteractive
+  class GitCommitter
+    def initialize(updater)
+      @updater = updater
+    end
+
+    def apply_updates_as_individual_commits(*gem_names)
+      assert_git_executable!
+      assert_working_directory_clean!
+
+      gem_names.flatten.each do |name|
+        updates = updater.apply_updates(name)
+        updated_gem = updates[name] || updates.values.first
+        next if updated_gem.nil?
```

Comment:
🗒️ `apply_updates` now returns a Hash of the gems that were actually updated. We use this to create a commit message that describes what gem was updated. There are some rare edge cases to consider:

1. Nothing was actually updated. This happens when bundler says "Bundler attempted to update [GEM] but its version stayed the same." This is because sometimes gems can't be updated individually and must be updated together. In this case we skip the gem and don't commit.
2. More than 1 gem was updated. For example, if we are updating `rails`, several other gems like `railties`, etc. are implicitly updated. For the commit message, we only care about the gem we explicitly asked for; `rails` in this case.
3. Gems other than the one we requested were updated. This is an edge case within edge case (1). Bundler might fail to update the gem we asked for, but still update one of the gem's sub-dependencies. I have seen this happen with `rubocop` and `rubocop-ast`. In this case, we will still create a commit; the commit message will contain the actual gem that was updated as opposed to the one we asked for.